app/views/users/show.html.erb

```diff
@@ -1,40 +1,59 @@
-<h2>Here is a list of posts for a given user</h2>
 <div class="wrapper">
-    <div class="userNameWrapper">
-        <div><img alt="Profile P."/></div>
-        <div class="userName">
-            <h2> 
-                Name: <%= @user.name %>
-            </h2>
-            <h6>Number of posts: <%= @user.posts.count%></h6>
-        </div>
+  <div class="userNameWrapper">
+    <div><img alt="Profile P."/></div>
+    <div class="userName">
+      <h2> Name: <%= @user.name %></h2>
+      <h6>Number of posts: <%= @user.posts.count %></h6>
     </div>
-        
-    <div class="bio">
-        <h3>Bio:</h3>
-        <p><%= @user.bio %></p>
-    </div>
-    <ul>
-        <% @user.posts.first(3).each_with_index do |post, index| %>
+  </div>
+
+  <div class="bio">
+    <h3>Bio:</h3>
+    <p><%= @user.bio %></p>
+  </div>
 
-          <ul class="post">
-            <%= link_to user_posts_path(@user), class: 'user-link' do %>
-                <h3>Post #<%= index + 1 %></h3>
-                <% words = post.text.split %>
-                <% if words.length <= 18 %>
-                <p><%= post.text %></p>
-                <% else %>
-                <p><%= words.take(18).join(' ') %>...</p>
-                <% end %>
-                <div class="count">
-                    Comments: <%= post.comments.count %> &nbsp;&nbsp;
-                    Likes: <%= post.likes.count %>
-                </div>
-            <% end %>
-          </ul>
+  <% if @all_posts %>
+    <% @all_posts.each_with_index do |post, index| %>
+      <ul class="post">
+        <%= link_to user_post_path(@user, post), class: 'user-link' do %>
+          <h3>Post #<%= index + 1 %></h3>
+          <% words = post.text.split %>
+          <% if words.length <= 18 %>
+            <p><%= post.text %></p>
+          <% else %>
+            <p><%= words.take(18).join(' ') %>...</p>
+          <% end %>
+          <div class="count">
+            Comments: <%= post.comments.count %> &nbsp;&nbsp;
+            Likes: <%= post.likes.count %>
+          </div>
         <% end %>
```

Comment:
- [ ] In the user show page we should be displaying the user's `recent_posts`, not all their posts. Kindly fix that.

Reply:
Well resolved ✔️